core/base/distanceField/DistanceField.h

```diff
@@ -54,7 +50,8 @@ namespace ttk {
       return 0;
     }
 
-    inline int setupTriangulation(Triangulation *triangulation) {
+    template <class TriangulationType = AbstractTriangulation>
+    inline int setupTriangulation(TriangulationType *triangulation) {
```

Comment:
you may want to rename that function into ```preconditionTriangulation``` (for consistency)

Reply:
also, for this one, I'd recommend not to template things (just use ```AbstractTriangulation```).
that function is supposed to be called only at initialization, so no need for advanced templating.

Reply:
but that's fine as it is (it'll use the default template parameter)